### ISSUE TITLE
[Bug] Ghost handles crash

### ISSUE THREAD

**Issue by l3td33r** [juzzlin/Heimer#150]
Steps to reproduce:
1. Hover over node so the handles appear
2. Quickly delete the node
3. The node is deleted, but the handles are not
4. The handles disappear after a few hundred milliseconds, but if we manage to click/grab one during this time, Heimer crashes

Example crash, child node creation handle:
```
Qt has caught an exception thrown from an event handler. Throwing
exceptions from an event handler is not supported in Qt.
You must not let any exception whatsoever propagate through Qt code.
If that is not possible, in Qt 5 you must at least reimplement
QCoreApplication::notify() and catch all exceptions there.

Invalid node index: 1
```

The error occurs only on the latest master https://github.com/juzzlin/Heimer/commit/52f0f987639d5d457ed97470c5edd8c1cc2e5ff6

**Comment by juzzlin** [comment]
Regression due to #143, should be relatively easy to fix.

**Comment by juzzlin** [comment]
Fixed in 75c5e9c76d061564b926e9ba501104e0d5d1f9f8
